### PR TITLE
#1128 - My Reimbursement Requests Endpoint

### LINKED ISSUE #1128: Finance - Get Reimbursement Requests For User
### Description

we want an endpoint that returns all the reimbursement requests for a single user

### Acceptance Criteria

- [x] New Get Receiver in the reimbursement routes file
  - Receiver will not have a body, calls controller function
  - Route should be on /reimbursement-requests/me
- [x] New Controller function called getUserReimbursementRequests in the reimbursement controller file
  - Gets the current user
  - Calls Service function
- [x] New service function called getUserReimbursementRequests in reimbursement services file
  - Function will filter out any deleted reimbursement request and then return the transformed reimbursement requests that the user created

- [x] Unit Tests

### Proposed Solution

^

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by jonathan-chen10**
## Changes

Add endpoint `/reimbursement-requests/current-user` to fetch all reimbursement requests whose recipientId matches the current user's.

### Jest tests

- Test that the appropriate Prisma function has been called with the desired inputs.

### Postman tests

Did not write seed data, instead I manually made the reimbursement requests like so.
Screen Shot 2023-06-04 at 1 47 31 PM
Screen Shot 2023-06-04 at 1 50 45 PM
Screen Shot 2023-06-04 at 1 52 47 PM

Database after inserting new reimbursement:
Screen Shot 2023-06-04 at 2 03 14 PM

- TEST CASE 1: Calling endpoint when Authorization = 1 (match found)
Screen Shot 2023-06-09 at 12 19 30 AM

- TEST CASE 2: Calling endpoint when Authorization = 2 (match not found)
Screen Shot 2023-06-09 at 12 19 51 AM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1128

### CODE REVIEW

File: src/backend/src/services/reimbursement-requests.services.ts [hunk 1]
```diff
@@ -25,6 +25,17 @@ import vendorTransformer from '../transformers/vendor.transformer';
 import sendMailToAdvisor from '../utils/transporter.utils';
 
 export default class ReimbursementRequestService {
+  /**
+   * Returns all reimbursement requests in the database that are created by a user.
+   * @param recipient the user retrieving their reimbursement requests
+   */
+  static async getUserReimbursementRequests(recipient: User): Promise<Reimbursement_Request[]> {
```
Comment (Peyton-McKee) [inline src/backend/src/services/reimbursement-requests.services.ts:32]: this should return a shared reimbursement request, you may need to merge megans branch into yours to get some of the shared types

File: src/backend/src/services/reimbursement-requests.services.ts [hunk 2]
```diff
@@ -25,6 +25,17 @@ import vendorTransformer from '../transformers/vendor.transformer';
...
+    const userReimbursementRequests = await prisma.reimbursement_Request.findMany({
+      where: { dateDeleted: null, recipientId: recipient.userId }
+    });
+    return userReimbursementRequests;
```
Comment (Peyton-McKee) [inline src/backend/src/services/reimbursement-requests.services.ts:36]: your gonna have to use the transformer here

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
Looks Great! Lmk if you have any questions about merging megans PR, also depending on when you want to work on it next we may be able to merge her PR first

File: src/backend/src/routes/reimbursement-requests.routes.ts [hunk 3]
```diff
@@ -13,6 +13,8 @@ const reimbursementRequestsRouter = express.Router();
 
 reimbursementRequestsRouter.get('/vendors', ReimbursementRequestController.getAllVendors);
 
+reimbursementRequestsRouter.get('/me', ReimbursementRequestController.getUserReimbursementRequests);
```
Comment (RChandler234) [inline src/backend/src/routes/reimbursement-requests.routes.ts:16]: let's do '/current-user' instead

File: src/backend/src/controllers/reimbursement-requests.controllers.ts [hunk 4]
```diff
@@ -9,6 +9,16 @@ import ReimbursementRequestService from '../services/reimbursement-requests.serv
 import { Vendor } from 'shared';
 
 export default class ReimbursementRequestsController {
+  static async getUserReimbursementRequests(_req: Request, res: Response, next: NextFunction) {
```
Comment (RChandler234) [inline src/backend/src/controllers/reimbursement-requests.controllers.ts:12]: rename to getCurrentUserReimbursementRequests

File: src/backend/src/services/reimbursement-requests.services.ts [hunk 5]
```diff
@@ -25,6 +25,17 @@ import vendorTransformer from '../transformers/vendor.transformer';
 import sendMailToAdvisor from '../utils/transporter.utils';
 
 export default class ReimbursementRequestService {
+  /**
+   * Returns all reimbursement requests in the database that are created by a user.
```
Comment (RChandler234) [inline src/backend/src/services/reimbursement-requests.services.ts:29]: reword to 'created by the given user'

File: src/backend/src/services/reimbursement-requests.services.ts (hunk 1 above)
Comment (RChandler234) [inline src/backend/src/services/reimbursement-requests.services.ts:32]: rename to getCurrentUserReimbursementRequests

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/backend/tests/test-data/reimbursement-requests.test-data.ts [hunk 6]
```diff
@@ -26,7 +30,7 @@ export const GiveMeMyMoney: PrismaReimbursementRequest = {
   dateOfExpense: new Date('18/8/2023'),
   recipientId: 1,
   vendorId: '',
-  account: ClubAccount.CASH,
+  account: Club_Accounts.CASH,
```
Comment (Peyton-McKee) [inline src/backend/tests/test-data/reimbursement-requests.test-data.ts:33]: can you just cahnge the test data to be ClubAccount

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
looks great after that one change it should be good and then we might merge this into megans and then merge megans @RChandler234 ?

File: src/backend/tests/test-data/reimbursement-requests.test-data.ts (hunk 6 above)
Comment (Peyton-McKee) [reply src/backend/tests/test-data/reimbursement-requests.test-data.ts:33]: nvm

**Review by Peyton-McKee — APPROVED** [review]
lgtm

**Comment by RChandler234** [comment]
> looks great after that one change it should be good and then we might merge this into megans and then merge megans @RChandler234 ?

yur

**Pull request merged**